pkg/splunk/util/util.go

```diff
@@ -139,7 +138,7 @@ func generateHECToken() []byte {
 }
 
 // PodExecCommand execute a shell command in the specified pod
-func PodExecCommand(c splcommon.ControllerClient, podName string, namespace string, cmd []string, stdin string, tty bool, mock bool) (string, string, error) {
+func PodExecCommand(c splcommon.ControllerClient, podName string, namespace string, cmd []string, streamOptions *remotecommand.StreamOptions, tty bool, mock bool) (string, string, error) {
```

Comment:
Just wondering what the particular reason was to move the `streamOptions` parameter to the callers responsibility?

Reply:
Good question. StreamOptions take the 'io.Reader' interface.  Existing code is taking string as stdin, which is converted as interface `strings.Reader`.  Now, with this API, we are passing the `PipeReader`. When I changed it to the generic `io.Reader`, it didn't go well, due to compatibility issues. Also, typecasting at the calling place didn't help.  So, felt this is the clean way, as the caller knows what will be the Stdin input.